x-pack/plugins/cases/server/client/metrics/alert_details.ts

```diff
@@ -5,33 +5,118 @@
  * 2.0.
  */
 
-import { CaseMetricsResponse } from '../../../common/api';
+import { merge } from 'lodash';
+
+import { AlertHostsMetrics, AlertUsersMetrics, CaseMetricsResponse } from '../../../common/api';
+import { createCaseError } from '../../common/error';
+import { AggregationFields, FrequencyResult, UniqueCountResult } from '../../services/alerts/types';
+import { CasesClient } from '../client';
+import { CasesClientArgs } from '../types';
 import { MetricsHandler } from './types';
 
 export class AlertDetails implements MetricsHandler {
-  /**
-   * This boolean protects against the metrics being queried multiple times. The applyMetrics function could be called
-   * once for each feature. All the metrics will be retrieved when the first applyMetrics is called though.
-   */
-  private retrievedMetrics: boolean = false;
+  private requestedFeatures: Set<AggregationFields> = new Set();
+
+  constructor(
+    private readonly caseId: string,
+    private readonly casesClient: CasesClient,
+    private readonly clientArgs: CasesClientArgs
+  ) {}
 
   public getFeatures(): Set<string> {
     return new Set(['alertHosts', 'alertUsers']);
   }
 
+  public setupFeature(feature: string) {
+    switch (feature) {
+      case 'alertHosts':
```

Comment:
keep in mind we changed the features to the dot notation: `alerts.hosts`, `alerts.users`

Reply:
Thanks for the reminder!